### PR TITLE
Update to PagerDuty API v2 

### PR DESCRIPTION
**Pull request by nadade**

### CODE REVIEW

**Comment by nadade** [comment]
[Change assign overlap to person who have conflict](https://trello.com/c/B36QEvO3/582-change-assign-overlap-to-person-who-have-conflict)

File: src/pagerduty-api.coffee [hunk 1]
```diff
@@ -2,26 +2,45 @@ _         = require 'underscore'
...
 request   = require 'request'
 nconf     = require 'nconf'
+url      = require 'url'
 
 # Factory for sending request to PD API
 send = (endpointPath, overrideOptions, cb) ->
   debug("Calling #{endpointPath} with options:", overrideOptions)
   sharedOptions =
-    uri: nconf.get('PAGERDUTY_API_URL') + endpointPath
+    uri: url.resolve 'https://api.pagerduty.com', endpointPath
```
Comment (miiila) [inline src/pagerduty-api.coffee]: I am bit confused now, why `PAGERDUTY_API_URL` is kept in config and readme.

**Comment by nadade** [comment]
addressed the comments so far

File: src/pagerduty.coffee [hunk 2]
```diff
@@ -119,18 +103,15 @@ processSchedules = (allSchedules, days = [], cb) ->
   duplicities = {}
   debug('allSchedules:', allSchedules)
   for schedule in allSchedules
-    debug('schedule:', schedule)
+#    debug('schedule:', JSON.stringify(schedule))
```
Comment (miiila) [inline src/pagerduty.coffee]: NIT: commented row; leftover?

File: test/fixtures/entries.json [hunk 3]
```diff
@@ -1,32 +1,80 @@
 {
-  "total": 3,
-  "entries": [
+  "total": null,
```
Comment (miiila) [inline test/fixtures/entries.json]: NIT: Why `null`? Maybe it's implied by PD Api, but I don't know by heart :-)

File: test/pagerduty-test.coffee [hunk 4]
```diff
@@ -5,6 +5,7 @@ debug    = require('debug')('pagerduty-overrides:tests')
 
 config   = require '../src/config'
 pd       = require '../src/pagerduty'
+pd_api   = require '../src/pagerduty-api'
```
Comment (miiila) [inline test/pagerduty-test.coffee]: NIT: One more camelCase

**Comment by miiila** [comment]
Leaving up to you to decide, whether you want to fix the NITs :-) Than please deploy it to AWS.

File: test/fixtures/entries.json (hunk 3 above)
Comment (nadade) [inline test/fixtures/entries.json]: @miiila 
```The total field is null by default. This behavior will provide the fastest response times when the total number of records is not important. If you need to know how many records match the query, pass a total=true parameter as part of the request and this field will be populated.```
(https://v2.developer.pagerduty.com/v2/docs/pagination)

Comment (miiila) [inline test/fixtures/entries.json]: 👍

**Comment by nadade** [comment]
resolves https://github.com/apiaryio/pagerduty-overlap-checker/issues/18

**Comment by nadade** [comment]
resolves #3

**Pull request merged**